packages/@repo/bundle-manager/src/helpers/__test__/versionUtils.test.ts

```diff
@@ -0,0 +1,186 @@
+import {describe, expect, it} from 'vitest'
+
+import {STALE_TAGS_EXPIRY_SECONDS} from '../../constants'
+import {currentUnixTime} from '../../utils'
+import {cleanupVersions, sortAndCleanupVersions} from '../versionUtils'
+
+describe('cleanupVersions()', () => {
+  it('applies cleanup logic when no new version specified', () => {
+    const now = currentUnixTime()
+    const versions = [
+      {timestamp: now - 100, version: '1.0.0'},
+      {timestamp: now - 200, version: '1.1.0'},
+      {timestamp: now - 300, version: '2.0.0'},
+    ]
+
+    const result = cleanupVersions(versions)
+    expect(result).toHaveLength(2) // One per major (highest semver)
+    expect(result.map((v) => v.version)).toEqual(['2.0.0', '1.1.0'])
+  })
+
+  it('applies TTL cleanup for new version major, keeps highest for other majors', () => {
```

Comment:
Any reason we couldn't keep other majors within the ttl limit here too? Might be missing something, but I don't see the harm, and it would simplify implementation a bit.